kernel/banner.c

```diff
@@ -30,6 +30,10 @@ void boot_banner(void)
 #endif /* defined(CONFIG_BOOT_DELAY) && (CONFIG_BOOT_DELAY > 0) */
 
 #if CONFIG_BOOT_BANNER
-	printk("*** Booting Zephyr OS build " BANNER_VERSION BANNER_POSTFIX " ***\n");
-#endif /* CONFIG_BOOT_BANNER */
+	if (strlen(CONFIG_BOOT_BANNER_OVERRIDE) > 0) {
```

Comment:
the `strlen()` is a no though, this means that both outputs will appear, should have Kconfig for overriding it and the string as a separate Kconfig entry